pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/UseUtilityClassRule.java

```diff
@@ -18,9 +18,9 @@
 import net.sourceforge.pmd.lang.java.ast.ASTMethodDeclaration;
 import net.sourceforge.pmd.lang.java.ast.ASTName;
 import net.sourceforge.pmd.lang.java.ast.ASTResultType;
-import net.sourceforge.pmd.lang.java.rule.AbstractLombokAwareRule;
+import net.sourceforge.pmd.lang.java.rule.AbstractJavaRule;
 
-public class UseUtilityClassRule extends AbstractLombokAwareRule {
+public class UseUtilityClassRule extends AbstractJavaRule {
```

Comment:
shouldn't we just extend `AbstractIgnoredAnnotationRule` and default it to `lombok.NoArgsConstructor`?

Reply:
@jsotuyod I've explained it in [comment](https://github.com/pmd/pmd/pull/958#discussion_r176766269). If I misunderstand it, please correct me:)

Reply:
@djydewang ok, that's compelling